.github/workflows/lint_test.yaml

```
@@ -0,0 +1,64 @@
+name: "CI"
+
+env:
+  FOUNDRY_PROFILE: "ci"
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - "main"
+      - "feature/**"
+      - "feat/**"
+      - "develop"
+  push:
+    branches:
+      - "main"
+      - "develop"
+
+jobs:
+  lint_and_test:
+    name: Lint & Test
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: "Check out the repo"
+        uses: "actions/checkout@v3"
+        with:
+          submodules: "recursive"
+
+      - name: "Install Foundry"
+        uses: "onbjerg/foundry-toolchain@v1"
+        with:
+          version: "nightly-d369d2486f85576eec4ca41d277391dfdae21ba7"
```

while we're at it, should we update to latest foundry release:
[nightly-fd87629fbc4ae2e0fa00ccf42b4a9ebe1b521d55](https://github.com/foundry-rs/foundry/releases/tag/nightly-fd87629fbc4ae2e0fa00ccf42b4a9ebe1b521d55)

```suggestion
          version: "nightly-fd87629fbc4ae2e0fa00ccf42b4a9ebe1b521d55"
```